From e2de6fab148e800380f1929fe4ea88a38de42053 Mon Sep 17 00:00:00 2001
From: "Alan T. DeKok" <aland@freeradius.org>
Date: Wed, 20 Nov 2019 13:59:54 -0500
Subject: [PATCH] a better fix for commit 30ffd21

Which still runs post-proxy-type fail if all of the home servers
are dead

[antorres@redhat.com: solved in FR 3.0.21, resolves bz#2030173]
[antorres@redhat.com: removed first hunk of commit, already present]
---
 src/main/process.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/src/main/process.c b/src/main/process.c
index c8b3af24e2..1a48517d43 100644
--- a/src/main/process.c
+++ b/src/main/process.c
@@ -2475,13 +2474,12 @@ static int process_proxy_reply(REQUEST *request, RADIUS_PACKET *reply)
 	}
 
 	old_server = request->server;
-	rad_assert(request->home_server != NULL);
 
 	/*
 	 *	If the home server is virtual, just run pre_proxy from
 	 *	that section.
 	 */
-	if (request->home_server->server) {
+	if (request->home_server && request->home_server->server) {
 		request->server = request->home_server->server;
 
 	} else {
@@ -3182,13 +3180,12 @@ do_home:
 	}
 
 	old_server = request->server;
-	rad_assert(request->home_server != NULL);
 
 	/*
 	 *	If the home server is virtual, just run pre_proxy from
 	 *	that section.
 	 */
-	if (request->home_server->server) {
+	if (request->home_server && request->home_server->server) {
 		request->server = request->home_server->server;
 
 	} else {
-- 
2.31.1

